src/libcxx.zig

```diff
@@ -223,8 +231,6 @@ pub fn buildLibCXX(comp: *Compilation, prog_node: std.Progress.Node) BuildError!
 
         if (std.mem.startsWith(u8, cxx_src, "src/support/win32/") and target.os.tag != .windows)
             continue;
-        if (std.mem.startsWith(u8, cxx_src, "src/support/solaris/") and !target.os.tag.isSolarish())
-            continue;
         if (std.mem.startsWith(u8, cxx_src, "src/support/ibm/") and target.os.tag != .zos)
             continue;
         if (!comp.config.any_non_single_threaded) {
```

Comment:
Not required, but I think you could move the handling of `src/support/win32/thread_win32.cpp` into your new `libcxx_thread_files` and remove the existing special casing of it here.

Reply:
- moved `thread_win32.cpp` to thread_files
- the skip filter is still needed for it but at least it's no longer checked for twice
- rebased